framework/src/base/Moose.C

```diff
@@ -261,14 +261,14 @@ addActionTypes(Syntax & syntax)
                            "(setup_mesh)"
                            "(add_mesh_generator)"
                            "(execute_mesh_generators)"
-                           "(recover_mesh_meta_data)"
                            "(set_mesh_base)"
                            "(check_copy_nodal_vars)"
                            "(add_partitioner)"
                            "(add_geometric_rm)"
                            "(attach_geometric_rm)"
                            "(init_mesh)"
                            "(prepare_mesh)"
+                           "(recover_mesh_meta_data)"
```

Comment:
Well, I thought I can just move this down bellow `set_mesh_base` but turns out I cannot. At that time, the element iterator does not provide the list of elements (empty), so the `construct` function in `dataLoad` is not functioning. I have to move this down further. Just to let you know.

Reply:
Well I remember why I put that task there. It's because it's at the equivalent spot when that data would be available during a normal run (e.g. right after the mesh generators are executed). It's not for a specific requirement. Moving it down is OK but it does create a small inconsistency in when data can be accessed, which is probably not a problem in practice. I don't love it, but I don't know if I have a better idea.